test/system/160-volumes.bats

```diff
@@ -64,6 +64,28 @@ function teardown() {
 }
 
 
+# Filter volumes by name
+@test "podman volume filter --name" {
+    suffix=$(random_string)
+    prefix="volume"
+
+    for i in 1 2; do
+        myvolume=${prefix}_${i}_${suffix}
+        run_podman volume create $myvolume
+        is "$output" "$myvolume" "output from volume create $i"
+    done
+
+    run_podman volume ls --filter name=${prefix}_1.+ --format "{{.Name}}"
+    is "$output" "${prefix}_1_${suffix}" "--filter name=${prefix}_1.+ shows only one volume"
+
+    run_podman volume ls --filter name=${prefix}* --format "{{.Name}}"
```

Comment:
Since you're adding _regex_, not glob, I would like to see this changed to `name=${prefix}_1*` with a comment loudly stating something like "yes, the '1' is intentional, blah blah this is regex". That will hammer the point to a casual reader who might be expecting globs.